arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/UberjarSimpleContainer.java

```diff
@@ -201,10 +204,16 @@ public void start(Archive<?> archive) throws Exception {
         // check for "org.wildfly.swarm.allDependencies" flag
         // see DependenciesContainer#addAllDependencies()
         if (archive instanceof DependenciesContainer) {
-            DependenciesContainer depContainer = (DependenciesContainer) archive;
-            if (depContainer.hasMarker("org.wildfly.swarm.allDependencies")) {
+            DependenciesContainer<?> depContainer =  (DependenciesContainer<?>) archive;
+            if (depContainer.hasMarker(DependenciesContainer.ALL_DEPENDENCIES_MARKER)) {
                 munge(depContainer, declaredDependencies);
             }
+        } else if (archive instanceof WebArchive) {
```

Comment:
We're still using `WebArchive` which does not implement `DependenciesContainer`. `WARArchive` is defined in undertow fraction and arquillian-adapter does not have this fraction as a dependency. We can add this dep and the code would be clearer, but I'm not sure it's acceptable.